brkraw/scripts/brkraw.py

```diff
@@ -466,5 +456,119 @@ def main():
+
+
+# This could integrate with cleanSubjectID, but mind the different warning messages
+def cleanSessionID(sess_id):
+    """To replace the underscore in session id.
+    Args:
+        sess_id (str): the orignal session id.
+    Returns:
+        str: the replaced session id.
+    """
+    # underscore will mess up bids output
+    if '_' in sess_id:
+        sess_id = sess_id.replace('_', 'Underscore')
+        import warnings
+        # warn user that the subject/participantID has a '_' and is replaced with 'Underscore'
+        warnings.warn('Session ID has "_"s, replaced with "Underscore" to make it bids compatiable')
+    return sess_id
+
+
+def generateModalityAgnosticFiles(root_path, json_fname):
+    """To create ModalityAgnosticFiles in output folder.
+    Args:
+        root_path (str): the root output folder
+        json_fname (str): I do not under why this variable is needed.
+    Returns:
+        nothing: just generate files.
+    """
+    data_des = 'dataset_description.json'
+    readme = 'README'
+    # why open use only the current folder and os.path not?
+    if not os.path.exists(data_des):
+        with open(os.path.join(root_path, 'dataset_description.json'), 'w') as f:
+            import json
+            import datetime
+            from ..lib.reference import DATASET_DESC_REF
+            json.dump(DATASET_DESC_REF, f, indent=4)
+    if not os.path.exists(readme):
+        with open(os.path.join(root_path, readme), 'w') as f:
+            # I do not know why json_fname here.
+            f.write('This dataset has been converted using BrkRaw (v{})'
+                    'at {}.\n'.format(json_fname, datetime.datetime.now()))
+            f.write('## How to cite?\n - https://doi.org/10.5281/zenodo.3818615\n')
+    
+    # https://bids-specification.readthedocs.io/en/stable/03-modality-agnostic-files.html
+    # participant.tsv file. if not exist, create it, and append. if need tab use \t
+    participantsTsvPath = os.path.join(root_path, 'participants.tsv')
+    if not os.path.exists(participantsTsvPath):
+        with open(participantsTsvPath, 'a+') as f:
+            f.write('participant_id\n')
+    else:
+        print('Exiting before convert..., participants.tsv already exist in output folder: ', participantsTsvPath)
+        sys.exit()
+
+    # participant.json file. if not exist, create it, and append. if need tab use \t
+    participantsJsonPath = os.path.join(root_path, 'participants.json')
+    if not os.path.exists(participantsJsonPath):
+        with open(participantsJsonPath, 'a+') as f:
+            sideCar = { 
+                "participant_id": {
+                    "Description": "Participant identifier"
+                }
+            }
+            json.dump(sideCar, f, indent=4)
+    else:
+        print('Exiting...before convert, participants.json already exist in output folder: ', participantsJsonPath)
+        sys.exit()
+
+
+
+def createFolderTree(multi_session, row, root_path, subj_code):
```

Comment:
These mkdir calls have a possibility of failing. I like creating a mkdirp function which creates if it doesn't exist:
```
def mkdirp(p):
    """Like mkdir -p"""
    path = os.path.join(p)
    try:
        os.makedirs(path)
    except OSError as exc:
        if exc.errno == errno.EEXIST:
            pass
        else:
            raise
	return path
```

Reply:
Here the `mkdir` refers a function defined in util file under lib, do you mean your function is better than this one (do you mind to explain a bit more why) and want to add that to the util file?
```
def mkdir(path):
    try:
        os.stat(path)
    except FileNotFoundError or OSError:
        os.makedirs(path)
    except:
        raise UnexpectedError
```

Reply:
Ah. I missed that this is already a custom mkdir command. I can do a seperate PR to slightly improve that existing implementation.

Reply:
Sure, that sounds great!